save-demo/src/main/kotlin/com/saveourtool/save/demo/controller/ManagementController.kt

```diff
@@ -5,89 +5,74 @@ import com.saveourtool.save.demo.DemoInfo
 import com.saveourtool.save.demo.DemoStatus
 import com.saveourtool.save.demo.entity.*
 import com.saveourtool.save.demo.service.*
-import com.saveourtool.save.demo.storage.DependencyStorage
-import com.saveourtool.save.domain.ProjectCoordinates
-import com.saveourtool.save.entities.FileDto
 import com.saveourtool.save.utils.*
 
 import org.springframework.http.HttpStatus
 import org.springframework.web.bind.annotation.*
-import reactor.core.publisher.Flux
 import reactor.core.publisher.Mono
 import reactor.kotlin.core.publisher.toMono
 import reactor.kotlin.core.util.function.component1
 import reactor.kotlin.core.util.function.component2
 
-import java.nio.ByteBuffer
-import java.time.LocalDateTime
-
-import kotlinx.datetime.toKotlinLocalDateTime
-
 /**
  * Internal controller that allows to create demos
  */
 @RestController
-@RequestMapping("/demo/internal")
+@RequestMapping("/demo/internal/manager")
 class ManagementController(
     private val toolService: ToolService,
     private val downloadToolService: DownloadToolService,
     private val demoService: DemoService,
-    private val dependencyStorage: DependencyStorage,
 ) {
     /**
      * @param demoDto
      * @return [Mono] of [DemoDto] entity
      */
-    @PostMapping("/add-tool")
-    fun addTool(@RequestBody demoDto: DemoDto): Mono<DemoDto> = demoDto.toMono()
+    @PostMapping("/add")
+    fun add(@RequestBody demoDto: DemoDto): Mono<DemoDto> = demoDto.toMono()
         .asyncEffect { downloadToolService.initializeGithubDownload(it.githubProjectCoordinates, it.vcsTagName) }
         .requireOrSwitchToResponseException({ validate() }, HttpStatus.CONFLICT) {
             "Demo creation request is invalid: fill project coordinates, run command and file name."
         }
-        .asyncEffect { blockingToMono { demoService.saveIfNotPresent(it.toDemo()) } }
+        .map { demoService.saveIfNotPresent(it.toDemo()).toDto() }
```

Comment:
better to use `flatMap { blockingToMono { demoService.saveIfNotPresent(it.toDemo()).toDto() } } `. I guess `saveIfNotPresent` is database operations and better to wrap by additional scheduler

Reply:
Oh, will do in next PR then

Reply:
https://github.com/saveourtool/save-cloud/pull/1852